client/src/app/pages/migration-targets/migration-targets.tsx

```diff
@@ -28,37 +28,74 @@ import { useTranslation } from "react-i18next";
 
 import { Item } from "./components/dnd/item";
 import { DndGrid } from "./components/dnd/grid";
-import { useFetchMigrationTargets } from "@app/queries/rulesets";
 import { CustomTargetForm } from "./custom-target-form";
+import { BundleOrderSetting, Setting } from "@app/api/models";
+import { updateBundleOrderSetting } from "@app/api/rest";
+import { AxiosPromise } from "axios";
+import {
+  BundleOrderSettingKey,
+  useFetchBundleOrder,
+  useFetchRuleBundles,
+} from "@app/queries/rulebundles";
 
 export const MigrationTargets: React.FC = () => {
   const { t } = useTranslation();
 
-  const { migrationTargets } = useFetchMigrationTargets();
+  const {
+    ruleBundles,
+    isFetching: isFetchingRuleBundles,
+    refetch: refetchRuleBundles,
+  } = useFetchRuleBundles();
 
-  const [activeId, setActiveId] = useState(null);
+  const {
+    bundleOrderSetting,
+    isFetching,
+    refetch: refreshBundleOrderSetting,
+  } = useFetchBundleOrder(ruleBundles);
 
-  const [targetIDs, setTargetIDs] = React.useState<string[]>(
-    migrationTargets.map((target) => target.name)
-  );
+  useEffect(() => {
+    refetchRuleBundles();
+    refreshBundleOrderSetting();
+  }, [isFetching]);
+
+  const [activeId, setActiveId] = useState(null);
 
   const [isCustomTargetFormOpen, setIsCustomTargetFormOpen] =
     React.useState(false);
 
   const sensors = useSensors(useSensor(MouseSensor), useSensor(TouchSensor));
 
-  function handleDragEnd(event: any) {
+  function handleDragOver(event: any) {
     const { active, over } = event;
 
     if (active.id !== over.id) {
-      setTargetIDs((items) => {
+      const reorderBundle = (items: number[]) => {
         const oldIndex = items.indexOf(active.id);
         const newIndex = items.indexOf(over.id);
 
         return arrayMove(items, oldIndex, newIndex);
-      });
+      };
+
+      const updatedBundleSetting: BundleOrderSetting = {
+        key: BundleOrderSettingKey,
+        value: reorderBundle(bundleOrderSetting.value),
+      };
+      let promise: AxiosPromise<Setting>;
+      if (updatedBundleSetting !== undefined) {
+        promise = updateBundleOrderSetting(updatedBundleSetting);
+      } else {
+        promise = updateBundleOrderSetting(updatedBundleSetting);
+      }
+      promise
+        .then((response) => {
+          refreshBundleOrderSetting();
+        })
+        .catch((error) => {});
     }
   }
+  useEffect(() => {
+    refreshBundleOrderSetting();
+  }, [refreshBundleOrderSetting]);
```

Comment:
Don't think the settings query will be able to be reused in its current state. Your settings query uses the  updateSetting 
axios call which converts the value of the request body to a string. That won't work for bundleOrder since we are storing it as an array of numbers.
``` 
export const updateSetting = (obj: Setting) =>
  axios.put<Setting>(`${SETTINGS}/${obj.key}`, obj.value?.toString());
```